### PR TITLE
Prevent some attributes from being merged with others on reexports

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Final fix for https://github.com/rust-lang/rust/issues/59368.

As discussed on zulip [here](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/Filtering.20sub.20attributes.20in.20ast.3A.3AAttribute), we need to clone the `Attribute` to be able to filter some parts of it. Then we need to go through the attributes to be able to only keep what we want (everything except a few attributes in short).

As for the second commit, when I wrote the test, I realized that the code to traverse all reexports one by one to collect all their attributes was not completely working so I fixed the few issues remaining.

r? @notriddle

### CODE REVIEW

**Review by notriddle — CHANGES_REQUESTED** [review]

File: src/librustdoc/clean/mod.rs [hunk 1]
```diff
@@ -2103,29 +2103,128 @@ fn get_all_import_attributes<'hir>(
...
+    use rustc_ast::token::{Token, TokenKind};
+    use rustc_ast::tokenstream::{TokenStream, TokenTree};
+
+    for attr in new_attrs {
+        let mut attr = attr.clone();
+        match attr.kind {
+            ast::AttrKind::Normal(ref mut normal) => {
+                if let [ident] = &*normal.item.path.segments {
+                    let ident = ident.ident.name;
+                    if ident == sym::doc {
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/clean/mod.rs:2177]: :scream: That's awesome!

**Comment by GuillaumeGomez** [comment]
Updated, thanks for the ideas!

File: src/librustdoc/clean/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/clean/mod.rs:2199]: I have the change locally. Weird. Well I'm fixing a bug discovered thanks to the `std` so it'll be part of next push in any case.

**Comment by GuillaumeGomez** [comment]
The last commit fixes the handling of reexports of `#[doc(hidden)]` items when the reexport doesn't have `#[doc(inline)]`.

**Pull request merged**